### PR TITLE
[2.x] Multiple snapshots

### LINKED ISSUE #871: `toMatchSnapshot` only supports one snapshot per test
It seems that `toMatchSnapshot` only supports one snapshot per test. I would have expected that either the index of the call or the name that is passed to the method will be used to generate the snapshot name.

### PR DESCRIPTION
**Pull request by fabio-ivona**
This PR will 

- [x] Cleanup `SnapshotRepository` code, as TestCase is retrieved inside the repository
- [x] Implement a snapshot expectations counter to keep track of multiple snapshots in the same test

**note** found an issue: when the first snapshot is created, a `$testCase::markTestIncomplete()` is called inside `toMatchSnapshot` expectation (Mixin/Expectation.php L837). This will stop the execution and prevent the creation of the second snapshot.

I commented out that line for now, resulting in a plain Risky test instead of an Incomplete one. Any idea on how to workaround on this?

### CODE REVIEW

File: src/Repositories/SnapshotRepository.php [hunk 1]
```diff
@@ -25,11 +24,9 @@ public function __construct(
     /**
      * Checks if the snapshot exists.
      */
-    public function has(TestCase $testCase, string $description): bool
+    public function has(): bool
```
Comment (fabio-ivona) [inline src/Repositories/SnapshotRepository.php:30]: @nunomaduro maybe we shall rename this one to `exists`?

File: src/Mixins/Expectation.php [hunk 2]
```diff
@@ -817,22 +823,18 @@ public function toMatchSnapshot(string $message = ''): self
...
                 $content,
                 $string,
                 $message === '' ? "Failed asserting that the string value matches its snapshot ($filename)." : $message
             );
         } else {
-            $filename = $snapshots->save($testCase, $string);
+            $filename = $snapshots->save($string);
 
-            $testCase::markTestIncomplete('Snapshot created at ['.$filename.'].');
+            //$testCase::markTestIncomplete('Snapshot created at ['.$filename.'].');
```
Comment (fabio-ivona) [inline src/Mixins/Expectation.php]: This one causes the test to end prematurely, preventing the creation of the second snapshot, I had to comment on it for now, any idea for a workaround?

File: src/Repositories/SnapshotRepository.php (hunk 1 above)
Comment (nunomaduro) [reply src/Repositories/SnapshotRepository.php:30]: @fabio-ivona Yes.

File: src/Mixins/Expectation.php (hunk 2 above)
Comment (nunomaduro) [reply src/Mixins/Expectation.php]: @freekmurze What do you do in your package here?

Comment (fabio-ivona) [reply src/Mixins/Expectation.php]: fixed

**Pull request merged**